admin/controller/extension/payment/razorpay.php

```diff
@@ -192,4 +194,1091 @@ protected function validate()
+            
+                );
+
+                $razorpay_plan = $api->plan->create($plan_data);
+                           
+                $this->log->write("RZP PlanID (:" . $razorpay_plan['id'] . ") created");
+            
+            }
+            catch(\Razorpay\Api\Errors\Error $e)
+            {
+                $this->log->write($e->getMessage());
+                $this->session->data['error'] = $e->getMessage();
+                echo "<div class='alert alert-danger alert-dismissible'> Something went wrong. Unable to create Razorpay Plan Id.</div>";
+                // $this->session->data['warning']= $e->getMessage();
+                // $this->response->redirect($this->url->link('extension/payment/razorpay/getPlan', 'user_token=' . $this->session->data['user_token'] . $url, true));
+                exit;
+            }
+            $this->model_extension_payment_razorpay->addPlan($this->request->post, $razorpay_plan['id']);
+            $this->load->model('localisation/language');
+
+            $languages= $data['languages'] = $this->model_localisation_language->getLanguages();
+            
+            // language_id
+            // print_r($languages); exit;
+            $this->model_extension_payment_razorpay->addRecurring($data);
+
+            $this->session->data['success'] = $this->language->get('text_plan_success');
+
+            $url = '';
+
+            if (isset($this->request->get['sort'])) {
+                $url .= '&sort=' . $this->request->get['sort'];
+            }
+
+            if (isset($this->request->get['order'])) {
+                $url .= '&order=' . $this->request->get['order'];
+            }
+
+            if (isset($this->request->get['page'])) {
+                $url .= '&page=' . $this->request->get['page'];
+            }
+
+            $this->response->redirect($this->url->link('extension/payment/razorpay/getPlan', 'user_token=' . $this->session->data['user_token'] . $url, true));
+        }
+
+        $this->getForm();
+    }
+   
+    //for status change
+    public function statusPlan()
+    {
+       
+        $this->load->language('extension/payment/razorpay');
+
+        $this->document->setTitle($this->language->get('heading_title'));
+
+        $this->load->model('extension/payment/razorpay');
+
+        if ((isset($this->request->post['selected'])) && ($this->request->post['status'])) {
+            $status = $this->request->post['status'];
+            if($status==1) {
+                foreach ($this->request->post['selected'] as $entity_id) {
+                    $this->model_extension_payment_razorpay->enablePlan($entity_id);
+                   
+                }
+
+                $this->session->data['success'] = $this->language->get('text_enable_success');
+            } else if($status==2) {
+                foreach ($this->request->post['selected'] as $entity_id) {
+                    $this->model_extension_payment_razorpay->disablePlan($entity_id);
```

Comment:
Make only one function to update status instead of 2 separate functions e.g., `updatePlanStatus($status, $entity_id)`